src/core/ext/filters/channel_idle/channel_idle_filter.cc

```diff
@@ -0,0 +1,379 @@
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <grpc/support/port_platform.h>
+
+#include <limits.h>
+#include <stdlib.h>
+
+#include <atomic>
+
+#include "src/core/ext/filters/channel_idle/idle_filter_state.h"
+#include "src/core/lib/channel/channel_args.h"
+#include "src/core/lib/channel/channel_stack_builder.h"
+#include "src/core/lib/channel/promise_based_filter.h"
+#include "src/core/lib/config/core_configuration.h"
+#include "src/core/lib/gprpp/capture.h"
+#include "src/core/lib/iomgr/timer.h"
+#include "src/core/lib/promise/exec_ctx_wakeup_scheduler.h"
+#include "src/core/lib/promise/loop.h"
+#include "src/core/lib/promise/sleep.h"
+#include "src/core/lib/promise/try_seq.h"
+#include "src/core/lib/transport/http2_errors.h"
+
+// TODO(juanlishen): The idle filter was disabled in client channel by default
```

Comment:
This comment does not appear to have changed.

Reply:
"Now the bug is fixed enable the filter by default." was the change... there's still a TODO needed here!

Reply:
Ah, I missed that change. Thought you might assign it to yourself since Juanli moved on. No issue.